ethicml/algorithms/inprocess/svm.py

```diff
@@ -5,17 +5,32 @@
 import pandas as pd
 
 from sklearn.svm import SVC
-from .in_algorithm import InAlgorithm
-from ..utils import DataTuple
+
+from ethicml.algorithms.inprocess.in_algorithm import InAlgorithm
+from ethicml.algorithms.utils import DataTuple
 
 
 class SVM(InAlgorithm):
     """Support Vector Machine"""
-    def run(self, train: DataTuple, test: DataTuple) -> pd.DataFrame:
+    def run(self, train: DataTuple, test: DataTuple, sub_process: bool = False) -> pd.DataFrame:
+        if sub_process:
+            return self.run_threaded(train, test)
```

Comment:
It's a bit ugly that this has to be repeated everywhere...

We could have a wrapper function like this:

```
def run(..., sub_process):
    if sub_process:
        self.run_threaded(...)
    else:
        self.__run(...)
```
and the real implementation would be in `__run()`.

Reply:
yeah, might need to have a think about what's the most friendly way of doing this. So if we decided to add FairGP, then what would be a reasonable set of methods to ask to be implemented